### PR TITLE
Add params for redis cluster provider

### PR DESCRIPTION
**Pull request by lucjross-favor**
This augments the Redis cluster provider to allow setting all the parameters supported by the backing RedisClusterContainer.

The only thing I don't use from the testcontainers definition is the setting of withKeyspaceNotifications because it uses a Docker image that doesn't seem to exist. The method in this PR seems better anyway because it allows the consumer to specify notification types.

### CODE REVIEW

**Comment by lucjross-favor** [comment]
Also, if someone could point me to where the docs live for Test Resources, I'm glad to update those.

**Review by melix — COMMENTED** [review]
LGTM! The docs are found in `src/main/docs/guide`, would be nice to add something indeed before merging.

**Comment by lucjross-favor** [comment]
OK, docs edited and would appreciate some proofreading

**Review by timyates — APPROVED** [review]

**Comment by lucjross-favor** [comment]
Thanks a bunch!

**Comment by lucjross-favor** [comment]
BTW I am not a code owner and cannot merge

**Pull request merged**
